static/src/js/components/GranuleResults/__tests__/GranuleResultsItem.test.js

```diff
@@ -335,316 +363,417 @@ function setup(type, overrideProps) {
 
-      const mockEvent = {
-        stopPropagation: jest.fn()
-      }
-
-      removeButton.simulate('click', mockEvent)
+      await user.click(screen.getByLabelText('Remove granule'))
       expect(props.onRemoveGranuleFromProjectCollection.mock.calls.length).toBe(1)
       expect(props.onRemoveGranuleFromProjectCollection.mock.calls[0]).toEqual([{
         collectionId: 'collectionId',
         granuleId: 'granuleId'
       }])
-
-      expect(mockEvent.stopPropagation.mock.calls.length).toBe(1)
-      expect(mockEvent.stopPropagation.mock.calls[0]).toEqual([])
     })
   })
 
   describe('download button', () => {
-    test('is passed the metrics callback', () => {
-      const { enzymeWrapper, props } = setup('cmr')
-      const dataLinksButton = enzymeWrapper.find(GranuleResultsDataLinksButton)
+    test('is passed the metrics callback', async () => {
+      const { props, user } = setup('cmr')
+
+      const { granule, onMetricsDataAccess, collectionId } = props
 
-      expect(dataLinksButton.props().onMetricsDataAccess).toEqual(props.onMetricsDataAccess)
+      const { dataLinks } = granule
+
+      const dataLink = dataLinks[0]
+
+      const { href } = dataLink
+
+      const dataLinksButton = await screen.findByRole('button', { name: 'Download single granule data' })
+
+      expect(dataLinksButton.href).toContain(href)
+
+      await act(async () => {
+        await user.click(dataLinksButton)
+      })
+
+      expect(onMetricsDataAccess).toHaveBeenCalledTimes(1)
+      expect(onMetricsDataAccess).toHaveBeenCalledWith(
+        expect.objectContaining({
+          type: 'single_granule_download',
+          collections: [{
+            collectionId
+          }]
+        })
+      )
     })
   })
 
   describe('download button with no link', () => {
     test('disables the button', () => {
-      const { enzymeWrapper } = setup('cmr-no-download')
-
-      const downloadButton = enzymeWrapper.find(GranuleResultsDataLinksButton)
+      setup('cmr-no-download')
 
-      expect(downloadButton.length).toEqual(0)
+      expect(screen.queryAllByLabelText('Download single granule data').length).toEqual(0)
     })
   })
 
   describe('without an granuleThumbnail', () => {
     test('does not render an granuleThumbnail', () => {
-      const { enzymeWrapper } = setup('no-thumb')
+      setup('no-thumb')
 
-      expect(enzymeWrapper.find('.granule-results-item__thumb').length).toEqual(0)
+      expect(screen.queryByLabelText('Link to granule')).not.toBeInTheDocument()
+      expect(screen.queryByLabelText('Granule thumbnail image')).not.toBeInTheDocument()
     })
 
     test('does not add the modifier class name', () => {
-      const { enzymeWrapper } = setup('no-thumb')
+      setup('no-thumb')
 
-      expect(enzymeWrapper.props().className).not.toContain('granule-results-item--has-thumbnail')
+      expect(screen.getByTestId('granule-results-item').className).not.toContain('granule-results-item--has-thumbnail')
```

Comment:
Avoid using test ids, need to find a way to find this element in a way that better simulates how a user would use the app